frontend/src/views/Storage/Storage.tsx

```diff
@@ -10,66 +10,88 @@ import {
-import { NMPFileFarmManureData } from '../../types';
-import { DEFAULT_NMPFILE_YEAR, DefaultManureFormData } from '../../constants';
 import { StyledContent } from './storage.styles';
-import useAppState from '../../hooks/useAppState';
 import { NUTRIENT_ANALYSIS, MANURE_IMPORTS } from '../../constants/routes';
 
 import { AppTitle, PageTitle, ProgressStepper, TabsMaterial } from '../../components/common';
 import { addRecordGroupStyle, customTableStyle, tableActionButtonCss } from '../../common.styles';
 import StorageModal from './StorageModal';
 
+export interface StorageForm {
+  ManureType: string;
+  ManureTypeName: string;
+  SystemName: string;
+  StorageName: string;
+  IsMaterialStored: boolean;
+  IsCovered: boolean;
+  UncoveredArea?: number;
+  AssignedToStoredSystem?: boolean;
+  ManagedManure: string;
+  UniqueMaterialName: string;
+}
+
+const EMPTY_STORAGE: StorageForm = {
+  ManureType: '',
+  ManureTypeName: '',
+  SystemName: '',
+  StorageName: '',
+  IsMaterialStored: false,
+  IsCovered: false,
+  AssignedToStoredSystem: false,
+  ManagedManure: '',
+  UniqueMaterialName: '',
+};
+
 export default function Storage() {
-  const { state } = useAppState();
   const navigate = useNavigate();
+  const [storageList, setStorageList] = useState<StorageForm[]>([]);
+  // Not sure if this is the right file type
+  const [storageFormData, setStorageFormData] = useState<StorageForm>(EMPTY_STORAGE);
 
-  // TODO: make correct file type
-  const [storageList, setStorageList] = useState<NMPFileFarmManureData[]>(
-    state.nmpFile.years[0]?.FarmManures || [],
-  );
-  // Not sure if NMPFileFarmManureData is the right file type
-  const [storageFormData, setstorageFormData] = useState<NMPFileFarmManureData>({
-    ManureSource: '',
-    MaterialType: '',
-    BookLab: '',
-    UniqueMaterialName: '',
-    Nutrients: { N: 0, P2O5: 0, K2O: 0, Moisture: '', NH4N: 0 },
-  });
-
-  const handleSubmit = () => {};
+  const handleSubmit = (formData: StorageForm) => {
+    setStorageList([...storageList, formData]);
+  };
 
   const handlePrevious = () => {
     navigate(MANURE_IMPORTS);
   };
 
+  // fix dispatch where are we saving the storage form info
+  // either imported or generated manure do we save IsMaterialStored and AssignedToStoredSystem booleans
   const handleNext = () => {
     navigate(NUTRIENT_ANALYSIS);
   };
 
   const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
 
   const handleDialogClose = () => {
+    setStorageFormData(EMPTY_STORAGE);
     setIsDialogOpen(false);
   };
 
+  // on edit text fields dont show up
   const handleEditRow = (e: GridRenderCellParams) => {
+    setStorageFormData(e.row);
     setIsDialogOpen(true);
   };
 
-  const handleDeleteRow = (e: GridRenderCellParams) => {};
+  const handleDeleteRow = (e: GridRenderCellParams) => {
+    setStorageList((prev) =>
+      prev.filter((row) => row.UniqueMaterialName !== e.row.UniqueMaterialName),
```

Comment:
I changed a lot of the app to use the actual row index for deletion. You should be able to do the same thing instead of filtering on the name.